esp/platform/espcontext.cpp

```diff
@@ -454,6 +454,15 @@ class CEspContext : public CInterface, implements IEspContext
             m_txSummary->append("total", m_processingTime, "ms");
         }
     }
+    virtual void addTraceSummaryCumulativeTime(const char* name, unsigned __int64 time)
+    {
+        if (m_txSummary)
+            m_txSummary->updateTimer(name, time);
```

Comment:
How frequently do you plan to call this function?  Is a hash table lookup each time likely to be a significant cost?